arch/x86/core/crt0.S

```diff
@@ -175,14 +175,23 @@ __csSet:
 #ifdef CONFIG_INIT_STACKS
 	movl $0xAAAAAAAA, %eax
 	leal _interrupt_stack, %edi
+#ifdef CONFIG_X86_STACK_PROTECTION
+	addl $4096, %edi
+#endif
 	stack_size_dwords = (CONFIG_ISR_STACK_SIZE / 4)
 	movl $stack_size_dwords, %ecx
 	rep  stosl
 #endif
 
 	movl	$_interrupt_stack, %esp
+#ifdef CONFIG_X86_STACK_PROTECTION
+	/* In this configuration, all stacks, including IRQ stack, are declared
+	 * with a 4K non-present guard page preceding the stack buffer
```

Comment:
Terminology nitpick: better to say "with the present bit unset in the PTE" or whatever.  Calling it "non-present" would lead a casual reader (me) to assume you meant something more like "unmapped" (i.e. that the memory wasn't real).  In fact the memory under the guard page is "present" in the sense of being physical memory, right?

The only reason I even stopped to think about the difference in meaning was that I was 82% sure you hadn't done any remapping yet and went looking to see if I missed a patch.